src/main/kotlin/parts/wisdom/arcraider/Coord.kt

```diff
@@ -0,0 +1,47 @@
+package parts.wisdom.arcraider
+
+import me.joypri.Mix
+import me.joypri.to
+import me.joypri.Part
+import me.joypri.Role
+
+object X : Role<Int>()
+object Y : Role<Int>()
+object Width: Role<Int>()
+object Height: Role<Int>()
+object Coords: Role<List<Coord>>()
+object TheGrid: Role<Grid>()
+
+open class Coord(vararg parts: Part) : Mix(*parts) {
+    val x by X
+    val y by Y
+}
+
+open class Grid(vararg parts:Part) : Mix(*parts) {
+    val width by Width
+    val height by Height
+    val coords by Coords
+
+    constructor(serializedGrid : SerializedGrid) :
+            this(
+                    Width to serializedGrid[0].size,
+                    Height to serializedGrid[1].size,
+                    Coords to generateCoords(serializedGrid)
+            ) {
+    }
+}
+
+// Not sure if this is a good place for this function to live. It turns a 2d array to a sparse list
+// of X, Y coorindates (ignores all black(empty) squares). This is probably also not the optimal way to
+// do this in kotlin. Diego will correct me :P
+fun generateCoords(serializedGrid: SerializedGrid): ArrayList<Coord> {
+    val coords = ArrayList<Coord>()
+    serializedGrid.forEach { row ->
+        row.forEach { col ->
+            if (row.indexOf(col) != 0) {
+                coords.add(Coord(X to row.indexOf(col), Y to serializedGrid.indexOf(row)))
+            }
+        }
+    }
+    return coords
```

Comment:
Hmm... I might not understand what we are trying to do here, but it looks like this function does not do what the comment says. 

I think the comment is saying we want to convert something like this:
```
[ 0, 1, 0 ]
[ 1, 1, 1 ]
[ 0, 0, 1 ] 
```
into something like this:
```
[0, 1], [1, 0], [1, 1], [2, 1], [2, 2]
```

If that's the case, then the check `if (row.indexOf(col) != 0) {` doesn't make sense, we are checking the index of the value rather than the value itself. What you actually want in this if is just `col != 0` since col is actually the values in the list and not the index (column).

I think what we want is this: 
```kotlin
val coords = mutableListOf<Coord>()
    serializedGrid.forEachIndexed { row, colValues ->
        colValues.forEachIndexed { col, value -> 
            if (value != 0) {
                coords.add(Coord(X to row, Y to col))
            }
        }
    }
```

Reply:
Also, I'm interpreting our input as [row, col] but we should check that it shouldn't be [col, row].